controllers/configurationpolicy_controller.go

```diff
@@ -335,15 +335,120 @@ func getTemplateConfigErrorMsg(err error) string {
+	isNamespaced bool
+}
+
+// getObjectTemplateDetails retrieves values from the object templates and returns an array of
+// objects containing the retrieved values.
+// It also gather namespaces for this policy if necessary:
+//   If a namespaceSelector is present AND objects are namespaced without a namespace specified
+func (r *ConfigurationPolicyReconciler) getObjectTemplateDetails(
+	plc policyv1.ConfigurationPolicy,
+) ([]objectTemplateDetails, []string, bool, error) {
+	templateObjs := make([]objectTemplateDetails, len(plc.Spec.ObjectTemplates))
+	selectedNamespaces := []string{}
+	queryNamespaces := false
+
+	for idx, objectT := range plc.Spec.ObjectTemplates {
+		var blob interface{}
+		if jsonErr := json.Unmarshal(objectT.ObjectDefinition.Raw, &blob); jsonErr != nil {
+			log.Error(jsonErr, "Could not unmarshal data from JSON")
+
+			return templateObjs, selectedNamespaces, false, jsonErr
+		}
+
+		var unstruct unstructured.Unstructured
+		unstruct.Object = make(map[string]interface{})
+		unstruct.Object = blob.(map[string]interface{})
+
+		templateObjs[idx].isNamespaced = r.isObjectNamespaced(&unstruct, true)
+
+		if objKind, ok := unstruct.Object["kind"]; ok {
+			templateObjs[idx].kind = strings.TrimSpace(objKind.(string))
+		}
+
+		if md, ok := unstruct.Object["metadata"]; ok {
+			metadata := md.(map[string]interface{})
+
+			if objNs, ok := metadata["namespace"]; ok {
+				templateObjs[idx].namespace = objNs.(string)
+			} else if templateObjs[idx].isNamespaced {
+				log.V(2).Info(
+					"metadata.namespace not found or could not be retrieved from the object template", "index", idx,
+				)
+			}
+
+			if objName, ok := metadata["name"]; ok {
+				templateObjs[idx].name = objName.(string)
+			} else {
+				log.V(2).Info(
+					"metadata.name not found or could not be retrieved from the object template", "index", idx,
+				)
+			}
+		}
+
+		if templateObjs[idx].isNamespaced && templateObjs[idx].namespace == "" {
+			queryNamespaces = true
+		}
+	}
+
+	// If required, query for namespaces specified in NamespaceSelector for objects to use
+	if queryNamespaces {
+		// Retrieve the namespaces based on filters in NamespaceSelector
+		selector := plc.Spec.NamespaceSelector
+		// If MatchLabels/MatchExpressions/Include were not provided, return no namespaces
+		if selector.MatchLabels == nil && selector.MatchExpressions == nil && len(selector.Include) == 0 {
+			log.Info("NamespaceSelector is empty. Skipping namespace retrieval.")
+		} else {
+			// If an error occurred in the NamespaceSelector, update the policy status and abort
+			var err error
+			selectedNamespaces, err = common.GetSelectedNamespaces(selector)
+			if err != nil {
+				log.Error(
+					err, "Error filtering namespaces with provided NamespaceSelector",
+					"namespaceSelector", fmt.Sprintf("%+v", selector))
+
+				reason := "NamespaceSelector error"
+				msg := fmt.Sprintf(
+					"Error filtering namespaces with provided NamespaceSelector: %s", err.Error())
```

Comment:
This is user-facing so the YAML casing would be better.
```suggestion
					"Error filtering namespaces with provided namespaceSelector: %s", err.Error())
```